File: languages/cpp/01-foundations/functions/example/function-overload-basics.cpp
// This helper example focuses on showing one focused overload or helper so the main example stays
// easy to scan.

#include <iostream>
#include <string>

using namespace std;

// Keep this helper separate so the main example can focus on the larger idea without extra noise.
void printValue(int value) { cout << "Integer value: " << value << '\n'; }

void printValue(double value) { cout << "Double value: " << value << '\n'; }

void printValue(const string& value) { cout << "String value: " << value << '\n'; }

int main() {
    printValue(42);
    printValue(3.14159);
    printValue("Hello from function overloads");

    return 0;
}

File: languages/csharp/01-foundations/functions/example/function-overload-basics.cs
// This helper example focuses on showing one focused overload or helper so the main example stays easy to scan.

// This extra example extends functions with overload-style dispatch.

using System;

// Keep this helper separate so the main example can focus on the larger idea without extra noise.
class Program
{
    static void PrintValue(int value)
    {
        Console.WriteLine($"Integer value: {value}");
    }

    static void PrintValue(double value)
    {
        Console.WriteLine($"Double value: {value}");
    }

    static void PrintValue(string value)
    {
        Console.WriteLine($"String value: {value}");
    }

    static void Main()
    {
        PrintValue(42);
        PrintValue(3.14159);
        PrintValue("Hello from function overloads");
    }
}
